### PR TITLE
Add Group SAML Links

### PR DESCRIPTION
**Pull request by mhodgson**
Gitlab Group SAML Group links added to production API yesterday: 
https://docs.gitlab.com/ee/api/groups.html#saml-group-links

### CODE REVIEW

**Comment by mhodgson** [comment]
Downstream Terraform Provider Update: https://github.com/gitlabhq/terraform-provider-gitlab/pull/1215

File: groups.go [hunk 1]
```diff
@@ -744,6 +753,92 @@ func (s *GroupsService) DeleteGroupLDAPLinkForProvider(gid interface{}, provider
 	return s.client.Do(req, nil)
 }
 
+// ListGroupSAMLLinks lists the group's SAML links. Available only for users who
+// can edit groups.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/groups.html#list-saml-group-links
+func (s *GroupsService) ListGroupSAMLLinks(gid interface{}, options ...RequestOptionFunc) ([]*SAMLGroupLink, *Response, error) {
```
[the remark this answers is not in the document]
Comment (mhodgson) [reply groups.go:761]: Nope, just confirmed with live testing in Postman.

**Comment by timofurrer** [comment]
I've just created a MR upstream on GitLab to improve a few things for this API, seehttps://gitlab.com/gitlab-org/gitlab/-/merge_requests/95561 - including:

* fixing a bug which lead to an empty response when creating a new SAML Group Link
* implemented a get endpoint to retrieve information for a single SAML Group Link

**Comment by timofurrer** [comment]
@mhodgson little update: my MR to the API from this morning was merged a few hours ago. Thus, I think you can implement it accordingly :)

**Comment by mhodgson** [comment]
@timofurrer I updated the code here to account for your upstream changes.

**Comment by timofurrer** [comment]
> @timofurrer awesome! Do you know when this would be in production? We're hoping to use this sooner than later...

@mhodgson It's already available on gitlab.com, see: https://docs.gitlab.com/ee/api/groups.html#get-saml-group-link

And it will come to the self-hosted version with GitLab 15.3 - which means that I'll try to get a release out of the terraform provider on either the 22nd or 23rd - right after 15.3 comes out.

**Review by timofurrer — APPROVED** [review]
Looks great 🎉

**Comment by mhodgson** [comment]
@timofurrer are you sure the changes are live? I'm testing the create endpoint and still getting a `null` response.

**Comment by timofurrer** [comment]
> @timofurrer are you sure the changes are live? I'm testing the create endpoint and still getting a `null` response. The GET endpoint also returns 404 for all requests, so thinking that isn't live either.

Interesting 🤔 I just assumed because the docs are live - it must be available "soon" ™️

**Comment by mhodgson** [comment]
@timofurrer looks like the updates are live! LGTM.

**Review by svanharmelen — APPROVED** [review]
This PR requires a few small tweaks which I will add in a separate commit...

**Comment by svanharmelen** [comment]
Thanks for your PR @mhodgson! Please see this PR for the last bits that you missed: https://github.com/xanzy/go-gitlab/commit/88c4146511b7582d41a1a38932e1bad38f43768a

**Pull request merged**